flakey-test-tracking/src/config.js

```
@@ -0,0 +1,32 @@
+// shouldTrackFlakes => tests are retried `numRetries` times and flakey results are logged w/ test output
+const shouldTrackFlakes =
+  (process.env.CIRCLECI && process.env.FLAKEY !== 'false') || process.env.FLAKEY === 'true'
+
+// numRetries === times test is run after the initial failure
+const numRetries = process.env.NUM_RETRIES ? Number(process.env.NUM_RETRIES) : 25
+
+// shouldSkipKnownFlakes => flakey test issues are fetched from github and corresponding tests are skipped
+const shouldSkipKnownFlakes =
+  shouldTrackFlakes && process.env.CIRCLECI && process.env.SKIP_KNOWN_FLAKES !== 'false'
```

Track the flakiness and skip them, could be separate things. I think that with that initial flag, you are adding a limit but maybe I could want a "fail fast without flaky". For example when you start with a migration, and you know that probably every test will fail, but I don't want to run it X times each one, but I want to already avoid the flaky ones. WDYT?

---

Hmm okay yea I like that. I'll add it

---

Actually, I think the way to do this would be to just set NUM_RETRIES=0 . Decoupling the <shouldTrackFlakes> logic and the shouldSkipTests logic would be non-trivial and I'm hesitant to make a change like that at this stage. What do you think about adding some documentation in the README explaining that if you want a 'fail fast without flaky' you should set NUM_RETRIES=0?

---

Ok, if it's documented, should be fine.
Another way could be to add a new flag, and if that flag is set, you could assume the NUM_RETRIES=0. But both seems good to me, your choice.